src/quota.rs

```diff
@@ -0,0 +1,29 @@
+use crate::grpc_sys::{self, grpc_resource_quota};
+use std::ptr;
+
+/// ResourceQuota represents a bound on memory and thread usage by the gRPC,
+/// Now grpc-rs only supports setting memory bound. A ResourceQuota can be
+/// consumed by a ChannelBuilder.
+pub struct ResourceQuota {
+    pub raw: *mut grpc_resource_quota,
+}
+
+impl ResourceQuota {
+    /// Create a control block for resource quota
+    pub fn new(name: Option<String>) -> ResourceQuota {
```

Comment:
Seems `Option<&str>` is enough. By the way, is empty string a valid argument?

Reply:
Yeah, and core will handle the valid string.